### ISSUE TITLE
Add support for file/media sharing

### ISSUE THREAD

**Issue by sirdrakeistda** [nextcloud/talk_matterbridge#38]
It should be possible to share also files/media trough matterbridge. It seems like Matterbridge already supports file sharing:
https://github.com/42wim/matterbridge/wiki/Mediaserver-setup-%28advanced%29

**Comment by gary-kim** [comment]
This is currently being worked on :).

**Comment by gary-kim** [comment]
There is support for downloading a file from Nextcloud here.

https://github.com/gary-kim/go-nc-talk/blob/720094d68ac9b09b77ef79f2e450ee1024a80c4c/user/user.go#L258-L272

An upload function still needs to be made. After that, support for files can be fully integrated into Matterbridge.

**Comment by gary-kim** [comment]
[Matterbridge v1.22.2](https://github.com/42wim/matterbridge/releases/tag/v1.22.2) includes support for sending file links to Nextcloud.

Work is going on for uploading files to Nextcloud here: https://github.com/gary-kim/go-nc-talk/pull/49

**Comment by sirdrakeistda** [comment]
> [Matterbridge v1.22.2](https://github.com/42wim/matterbridge/releases/tag/v1.22.2) includes support for sending file links to Nextcloud.
> 
> Work is going on for uploading files to Nextcloud here: [gary-kim/go-nc-talk#49](https://github.com/gary-kim/go-nc-talk/pull/49)
File transfer does not work for me yet with talk_matterbridge (Version 1.22.3). (I'm not sure if file transfer is supposed to work)

**Comment by gary-kim** [comment]
Currently, sending files from Talk should work fully and files sent to Talk should be sent links if they are available.

**Comment by webagroprom** [comment]
There are positive changes in this task ?

**Comment by gary-kim** [comment]
Yes, at this point, there is full support for file uploads in the [go-nc-talk](https://github.com/gary-kim/go-nc-talk) library that provides the Talk support in Matterbridge. Just need to make a PR for Matterbridge to use those features now.
